### PR TITLE
fix: y-axis no longer cuts off digits, instead uses SI format with 2 significant digits

### PR DESCRIPTION
**Pull request by corteggiano**
## Overview
fix y-axis values so they are not cropped off -- [view issue for more details](https://github.com/awslabs/synchro-charts/issues/66)

## Tests
[Include a link to the passing GitHub action running the test suite here.]

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/synchro-charts/src/components/charts/sc-webgl-base-chart/renderAxis.ts [hunk 1]
```diff
@@ -68,8 +68,9 @@ const xAxisConstructor = (size: SizeConfig, viewport: ViewPort) => {
 const yAxisConstructor = (size: SizeConfig, viewport: ViewPort) => {
   const { yTickCount } = tickCount(size);
   const { yScale } = scales(size, viewport);
+
   return axisLeft(yScale)
-    .ticks(yTickCount)
+    .ticks(yTickCount, 's')
```
Comment (boweihan) [inline packages/synchro-charts/src/components/charts/sc-webgl-base-chart/renderAxis.ts:73]: Cool :). Looks like d3-format has a whole bunch of options - https://github.com/d3/d3-format#locale_formatPrefix. 

Is there a reason you chose this particular format vs. exponential which is used on chart tooltips? (https://github.com/awslabs/synchro-charts/issues/66)

**Review by boweihan — CHANGES_REQUESTED** [review]
I love one liners!

Looks like the build is failing on some unit tests - we should make sure that we have some test coverage on this change.

File: packages/synchro-charts/src/components/charts/sc-webgl-base-chart/renderAxis.ts (hunk 1 above)
Comment (corteggiano) [reply packages/synchro-charts/src/components/charts/sc-webgl-base-chart/renderAxis.ts:73]: exponential feels significantly harder to read, while decimal notation with SI prefix is something most of us are used to seeing 
ie: `50k` vs `5.0e+4`

File: packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-axis.spec.tsx [hunk 2]
```diff
@@ -126,11 +126,11 @@ describe('initial render', () => {
 
   it('has expected y axis value labels', async () => {
     const { axis } = await axisSpecPage();
-    expect(axis.innerHTML).toContain('3,000');
-    expect(axis.innerHTML).not.toContain('3,025');
-    expect(axis.innerHTML).toContain('3,050');
+    expect(axis.innerHTML).toContain('3.00k');
+    expect(axis.innerHTML).not.toContain('3.0k');
```
Comment (boweihan) [inline packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-axis.spec.tsx:130]: does 3.025 get rounded down?

File: packages/synchro-charts/cypress/integration/charts/line-chart/draggable-annotations.spec.ts [hunk 3]
```diff
@@ -240,10 +240,10 @@ it('draggable annotation rescales axis in negative direction', () => {
   cy.get('g.axis.y-axis > g.tick').should('have.length', 10);
   cy.get('g.axis.y-axis > g.tick > text')
     .eq(9)
-    .should('have.text', '5,000');
+    .should('have.text', '5k');
```
Comment (boweihan) [inline packages/synchro-charts/cypress/integration/charts/line-chart/draggable-annotations.spec.ts:243]: is the number of significant digits based on the overall y-axis range?

File: packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-axis.spec.tsx (hunk 2 above)
Comment (corteggiano) [reply packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-axis.spec.tsx:130]: yeah, this is an example from 2025 to 3025 and they both round down 
image

File: packages/synchro-charts/cypress/integration/charts/line-chart/draggable-annotations.spec.ts (hunk 3 above)
Comment (corteggiano) [reply packages/synchro-charts/cypress/integration/charts/line-chart/draggable-annotations.spec.ts:243]: yep! if the values are greater than 10k apart it wont add in the significant digits

examples: `0 -> 6000`
image
`7000 -> 12000`
image
`7000 -> 16500`
image
`7000 -> 17000`
image

**Comment by diehbria** [comment]
squash commits and make single commit following conventional commit (guidelines found at https://github.com/awslabs/iot-app-kit/blob/main/docs/CodingGuidelines.md)

File: packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-axis.spec.tsx [hunk 4]
```diff
@@ -126,11 +126,10 @@ describe('initial render', () => {
 
   it('has expected y axis value labels', async () => {
     const { axis } = await axisSpecPage();
-    expect(axis.innerHTML).toContain('3,000');
-    expect(axis.innerHTML).not.toContain('3,025');
-    expect(axis.innerHTML).toContain('3,050');
+    expect(axis.innerHTML).toContain('3.0k');
+    expect(axis.innerHTML).toContain('3.1k');
+    expect(axis.innerHTML).not.toContain('3.00k');
     expect(axis.innerHTML).not.toContain('3,075');
```
Comment (boweihan) [inline packages/synchro-charts/src/components/charts/sc-webgl-base-chart/sc-webgl-axis.spec.tsx:132]: Does it make sense to keep this particular test case?

**Review by boweihan — APPROVED** [review]
LGTM!

**Pull request merged**